packages/state-transition/src/util/loadState/loadValidator.ts

```diff
@@ -0,0 +1,42 @@
+import {CompositeViewDU, ValueOf} from "@chainsafe/ssz";
+import {deserializeContainerIgnoreFields, ssz} from "@lodestar/types";
+
+/**
+ * Load validator from bytes given a seed validator.
+ * - If it's the same validator, reuse pubkey and withdrawal credentials to save memory
+ * - If it's a new validator, deserialize it
+ */
+export function loadValidator(
+  seedValidator: CompositeViewDU<typeof ssz.phase0.Validator>,
+  newValidatorBytes: Uint8Array
+): CompositeViewDU<typeof ssz.phase0.Validator> {
+  if (isSameValidator(seedValidator, newValidatorBytes)) {
```

Comment:
Extend this checks for the 3 cases: same pk, same withdrawals_creds, and same pk + same withdrawals_creds